aws/table_aws_config_rule.go

```diff
@@ -0,0 +1,220 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/configservice"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsConfigRule(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_config_rule",
+		Description: "AWS Config Rule",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("name"),
+			ShouldIgnoreError: isNotFoundError([]string{"NoSuchConfigRuleException", "ResourceNotFoundException"}),
+			Hydrate:           getConfigRule,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listConfigRules,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name that you assign to the AWS Config rule.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ConfigRuleName"),
+			},
+			{
+				Name:        "rule_id",
+				Description: "The ID of the AWS Config rule.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ConfigRuleId"),
+			},
+			{
+				Name:        "rule_arn",
```

Comment:
```suggestion
				Name:        "arn",
```

@ParthaI can you please update this column name here, in the docs, and in the integration tests?

Reply:
# Integration test logs
<details>
  <summary>Logs</summary>

```
SETUP: tests/aws_config_rule []

PRETEST: tests/aws_config_rule

TEST: tests/aws_config_rule
Running terraform
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
data.aws_region.alternate: Refreshing state...
aws_config_config_rule.named_test_resource: Creating...
aws_config_config_rule.named_test_resource: Creation complete after 3s [id=turbottest74522]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
region_name = us-east-1
resource_aka = arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk
resource_name = turbottest74522
rule_id = config-rule-hkw5fk

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk"
    ],
    "arn": "arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk",
    "description": null,
    "name": "turbottest74522",
    "rule_id": "config-rule-hkw5fk",
    "rule_state": "ACTIVE",
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest74522"
      }
    ],
    "title": "turbottest74522"
  }
]
✔ PASSED

Running SQL query: test-list-call-query.sql
[
  {
    "arn": "arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk",
    "name": "turbottest74522",
    "rule_id": "config-rule-hkw5fk",
    "title": "turbottest74522"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk"
    ],
    "arn": "arn:aws:config:us-east-1:986325076436:config-rule/config-rule-hkw5fk",
    "tags": {
      "name": "turbottest74522"
    },
    "title": "turbottest74522"
  }
]
✔ PASSED

POSTTEST: tests/aws_config_rule

TEARDOWN: tests/aws_config_rule

SUMMARY:

1/1 passed.

```

</details>

Reply:
Updated the column name `rule_arn` to `arn`